includes/kco-functions.php

```diff
@@ -680,6 +680,51 @@ function kco_validate_order_total( $klarna_order, $order ) {
 	return true;
 }
 
+/**
+ * Validate the Klarna Checkout item quantities against the Woo order.
+ *
+ * @param array    $klarna_order The Klarna order.
+ * @param WC_Order $order The Woo order.
+ *
+ * @return bool
+ */
+function kco_validate_cart_content( $klarna_order, $order ) {
+	foreach ( $order->get_items() as $item ) {
+		// Only check quantity for product items.
+		if ( 'line_item' !== $item->get_type() ) {
+			continue;
+		}
+
+		// Assume the item is a mismatch until proven otherwise. This way, in the unlikely chance of not finding any matching item, we can set the order to on-hold.
+		$mismatch = true;
+
+		// Try the specific (variation id) before the general (product id).
+		$product_id = 0 !== $item->get_variation_id() ? $item->get_variation_id() : $item->get_product_id();
+		$product    = wc_get_product( $product_id );
```

Comment:
The item in this case should have a method called [get_product](https://woocommerce.github.io/code-reference/classes/WC-Order-Item-Product.html#method_get_product) which should be able to replace this.